cmd/juju/backups/doc.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package backups provides the juju backups command.
+// Backup of juju's state is a critical feature, not only for juju users
+// but for use inside juju itself. This is likewise the case for the
+// ability to restore previous backups.
+
+// Backing up juju state involves dumping the state database and copying all
+// files that are critical to juju's operation. All the files are bundled up
+// into an archive file. Effectively the archive represents a snapshot of juju state.
+
+// First of all, running juju commands will fail while the DB is
+// unavailable (already running services should not be affected). While
+// this period of time is rather short, we expect that it will grow over
+// time. Furthermore, the larger a model's state, the larger the
+// impact of this downtime. In the long term this makes it less than ideal
+// to run backup as often as it should be.
+
+// Secondly, state currently does not block for the backup process as a
+// whole. This means that if we dump the DB first, it may be out of date
+// by the time we finish gathering the state-related files. In practice
+// this isn't a big concern since we do not expect the files to change
+// during the interval that backup is running. However, we do backup some
+// log files, so there is a small chance they will differ from when backups
+// started.
+
+// HA is a work in progress, for the moment we have a basic support which is an
+// extension of the regular backup functionality.
+// Read carefully before attempting backup/restore on an HA model.
+
+// In the case of HA, the backup process will backup files/db for machine 0,
+// support for "any working controller" is plans for the near future.
+// We assume, for now, that if you are running restore is because you have
+// lost all controller machines. Out of this restore you will get one
+// functioning controller that you can use to start you other state machines.
+// BEWARE, only run restore in the case where you no longer have working
+// controllers since otherwise this will take them offline and possibly
+// cripple your model.
+
+// Each backup archive is a gzipped tar file (.tar.gz).  It has the
```

Comment:
This following content is better. We would want to add info about how the controller creates the backup archive and then streams it back to the client so the archive ends up on the local user's disk etc.